src/app/ui/user-profile/user-profile.component.html

```diff
@@ -24,15 +24,16 @@ <h3>Hello, GUEST</h3>
     <!-- template will replace this div -->
   </div>
 
-  <div class="box" *ngIf="user.team?.lead === this.user.uid && !player">
+  <div class="box" *ngIf="this.user.uid === player.uid || isLead">
     <mat-card class="example-card">
       <mat-card-actions>
-        <button mat-stroked-button color="primary"><a routerLink="/profile" [queryParams]="{uid: 'Y2FlYmU0YmItMTkyYi00NGQwLTljYmQtYjEyODJmNjVjZTdl.NDU3ZDU2ODUtMDQ2Ny00ZDA1LWIyM2ItOGY4MTdhZGRhNDdj'}">View Profile</a></button> 
+        <button *ngIf="isLead" mat-stroked-button color="primary"><a routerLink="/profile" [queryParams]="{uid: 'player.uid'}">View Profile</a></button> 
+        <button mat-stroked-button color="primary">Other Functionality</button>
       </mat-card-actions>
     </mat-card>
   </div>
 
-  <div class="box" *ngIf="user.team?.lead === this.user.uid && player">
+  <div class="box" *ngIf="isLead && player">
```

Comment:
need to consult you guys here, I really wasn't sure what the if conditions are supposed to achieve in the bigger picture, sorry guys. Temporarily, profile can only be viewed if u r either team Lead, or it's your own profile. Add quest can only be given if you are a lead.